src/execution/debug.ts

```diff
@@ -105,17 +105,15 @@ export class GaugeDebugger {
         }
     }
 
-    public startDebugger(): Promise<any> {
-        return new Promise((res, rej) => {
-            setTimeout(() => {
-                let folder = workspace.getWorkspaceFolder(window.activeTextEditor.document.uri);
-                let root = this.clientsMap.get(window.activeTextEditor.document.uri.fsPath).project.root();
-                if (!folder) {
-                    rej(`The debugger dones not work for a stand alone file.Please open the folder ${root}.`);
-                }
-                debug.startDebugging(folder, this.getDebuggerConf()).then(res, rej);
+    public startDebugger(): void {
+        let folder = workspace.getWorkspaceFolder(window.activeTextEditor.document.uri);
+        let root = this.clientsMap.get(window.activeTextEditor.document.uri.fsPath).project.root();
+        if (!folder) {
+            throw new Error(`The debugger does not work for a stand alone file. Please open the folder ${root}.`);
+        }
+        setTimeout( () => {
+            debug.startDebugging(folder, this.getDebuggerConf());
```

Comment:
` debug.startDebugging` is an `async` function, shouldn't we await for it. The previous implementation used to resolve it. That's the reason it used to return a promise.

Reply:
The async function was the reason the ruby debugger was quitting, we are not sure why, adding those promises breaks the debugger. It doesn't connect properly.